packages/adapters-library/src/defiProvider.ts

```diff
@@ -143,7 +143,10 @@ export class DefiProvider {
 
     const userPoolsByChain = (
       await filterMapAsync(Object.values(EvmChain), async (chainId) => {
-        if (filterChainIds && !filterChainIds.includes(chainId)) {
+        if (
+          filterProtocolTokens ||
```

Comment:
If a list of protocol tokens is passed, we don't need the user events as they won't be used.